### PR TITLE
[WIP] v1.0.0-rc.1

### PR DESCRIPTION
**Pull request by bartlomieju**
Waiting on #4933

Tentatively set versions to `1.0.0-rc.1` but I'm pretty sure we don't want that for `deno_core`, `deno_typescript`

### CODE REVIEW

**Comment by nayeemrmn** [comment]
Is `std` going 1.0 as well? I feel like the release cycles should split now... otherwise someone should go through and underscore all private modules.

File: core/Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 [package]
 name = "deno_core"
-version = "0.41.0"
+version = "1.0.0-rc.1"
```
Comment (ry) [inline core/Cargo.toml:5]: 0.42.0 (or 0.41.1 .. not sure if we made any API changes here)

**Comment by bartlomieju** [comment]
Going with `v0.42.0` instead in #4986

**Comment by ry** [comment]
(We will do 1.0 rc1 in a few days - but due to the number of API changes we want to make another point release first and potentially address any issues.)

**Pull request closed without merging**